### ISSUE TITLE
t/diagnostics/plugin-fail.t fails (with Moose 2.1806)

### ISSUE THREAD

**Issue by eserte** [rjbs/Dist-Zilla#567]
My smoker machines started to report the following failure:
```
    #   Failed test 'Exception explains that it couldn't load the plugin 2-layers down'
    #   at t/diagnostics/plugin-fail.t line 46.
    #                   'You can only consume roles, Some::Package::That::Does::Not::Exist::Due::To::A::Typo is not a Moose role at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Moose/Exporter.pm line 419
    # 	Moose::with('Some::Package::That::Does::Not::Exist::Due::To::A::Typo') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/t/lib/Dist/Zilla/Plugin/BrokenPlugin4.pm line 8
    # 	require Dist/Zilla/Plugin/BrokenPlugin4.pm at /usr/perl5.24.0p/lib/site_perl/5.24.0/Module/Runtime.pm line 317
    # 	Module::Runtime::require_module('Dist::Zilla::Plugin::BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 174
    # 	Class::Load::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 99
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x48006bb8)', 'Try::Tiny::Catch=REF(0x48006348)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 180
    # 	Class::Load::try_load_class('Dist::Zilla::Plugin::BrokenPlugin4', undef) called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 118
    # 	Class::Load::load_optional_class('Dist::Zilla::Plugin::BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 231
    # 	Config::MVP::Section::load_package('Dist::Zilla::MVP::Section=HASH(0x47fc2510)', 'Dist::Zilla::Plugin::BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 277
    # 	Config::MVP::Section::_BUILD_package_settings('Dist::Zilla::MVP::Section=HASH(0x47fc2510)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 287
    # 	Config::MVP::Section::BUILD('Dist::Zilla::MVP::Section=HASH(0x47fc2510)', 'HASH(0x48006960)') called at constructor Dist::Zilla::MVP::Section::new (defined at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/MVP/Section.pm line 44) line 89
    # 	Dist::Zilla::MVP::Section::new('Dist::Zilla::MVP::Section', 'HASH(0x48006a68)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 144
    # 	Config::MVP::Assembler::begin_section('Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)', 'BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 187
    # 	Config::MVP::Assembler::change_section('Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)', 'BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/INI.pm line 53
    # 	Config::MVP::Reader::INI::INIReader::change_section('Config::MVP::Reader::INI::INIReader=HASH(0x48006390)', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/INI/Reader.pm line 111
    # 	Config::INI::Reader::read_handle('Config::MVP::Reader::INI::INIReader=HASH(0x48006390)', 'IO::File=GLOB(0x480062e8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Mixin/Linewise/Readers.pm line 103
    # 	Mixin::Linewise::Readers::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/INI.pm line 25
    # 	Config::MVP::Reader::INI::read_into_assembler('Config::MVP::Reader::INI=HASH(0x480060d8)', '/home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/tmp/mkPAbY6Wau/source/dist.ini', 'Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader.pm line 53
    # 	Config::MVP::Reader::read_config('Config::MVP::Reader::INI=HASH(0x480060d8)', '/home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/tmp/mkPAbY6Wau/source/dist.ini', 'HASH(0x47fc2f90)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/Finder.pm line 115
    # 	Config::MVP::Reader::Finder::read_config('Dist::Zilla::MVP::Reader::Finder', 'Path::Tiny=ARRAY(0x47fc2fd8)', 'HASH(0x47fc2f90)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 287
    # 	Dist::Zilla::Dist::Builder::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 103
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x47fc2e58)', 'Try::Tiny::Catch=REF(0x41cc4798)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 312
    # 	Dist::Zilla::Dist::Builder::_load_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47fc2498)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 45
    # 	Dist::Zilla::Dist::Builder::from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Tester.pm line 196
    # 	Class::MOP::Class:::around('CODE(0x4510e810)', 'Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Class/MOP/Method/Wrapped.pm line 164
    # 	Dist::Zilla::Tester::_Builder::_wrapped_from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Class/MOP/Method/Wrapped.pm line 95
    # 	Dist::Zilla::Tester::_Builder::from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at t/diagnostics/plugin-fail.t line 13
    # 	main::mkconfig('corpus/dist/DZT', 'ARRAY(0x41cc46c0)') called at t/diagnostics/plugin-fail.t line 43
    # 	main::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Fatal.pm line 129
    # 	Test::Fatal::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 101
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x41cc4270)', 'Try::Tiny::Catch=REF(0x47fc1be8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Fatal.pm line 136
    # 	Test::Fatal::exception('CODE(0x41cfe4f8)') called at t/diagnostics/plugin-fail.t line 43
    # 	main::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Builder.pm line 300
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Builder.pm line 300
    # 	Test::Builder::subtest('Test::Builder=HASH(0x41d08540)', 'BrokenPlugin4', 'CODE(0x41cfe480)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/More.pm line 807
    # 	Test::More::subtest('BrokenPlugin4', 'CODE(0x41cfe480)') called at t/diagnostics/plugin-fail.t line 48
    # Compilation failed in require at /usr/perl5.24.0p/lib/site_perl/5.24.0/Module/Runtime.pm line 317, <GEN3> line 8. at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 144.
    # '
    #     doesn't match '(?^:Can't locate)'

    #   Failed test 'Exception reports the original problem'
    #   at t/diagnostics/plugin-fail.t line 47.
    #                   'You can only consume roles, Some::Package::That::Does::Not::Exist::Due::To::A::Typo is not a Moose role at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Moose/Exporter.pm line 419
    # 	Moose::with('Some::Package::That::Does::Not::Exist::Due::To::A::Typo') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/t/lib/Dist/Zilla/Plugin/BrokenPlugin4.pm line 8
    # 	require Dist/Zilla/Plugin/BrokenPlugin4.pm at /usr/perl5.24.0p/lib/site_perl/5.24.0/Module/Runtime.pm line 317
    # 	Module::Runtime::require_module('Dist::Zilla::Plugin::BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 174
    # 	Class::Load::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 99
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x48006bb8)', 'Try::Tiny::Catch=REF(0x48006348)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 180
    # 	Class::Load::try_load_class('Dist::Zilla::Plugin::BrokenPlugin4', undef) called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Class/Load.pm line 118
    # 	Class::Load::load_optional_class('Dist::Zilla::Plugin::BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 231
    # 	Config::MVP::Section::load_package('Dist::Zilla::MVP::Section=HASH(0x47fc2510)', 'Dist::Zilla::Plugin::BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 277
    # 	Config::MVP::Section::_BUILD_package_settings('Dist::Zilla::MVP::Section=HASH(0x47fc2510)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Section.pm line 287
    # 	Config::MVP::Section::BUILD('Dist::Zilla::MVP::Section=HASH(0x47fc2510)', 'HASH(0x48006960)') called at constructor Dist::Zilla::MVP::Section::new (defined at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/MVP/Section.pm line 44) line 89
    # 	Dist::Zilla::MVP::Section::new('Dist::Zilla::MVP::Section', 'HASH(0x48006a68)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 144
    # 	Config::MVP::Assembler::begin_section('Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)', 'BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 187
    # 	Config::MVP::Assembler::change_section('Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)', 'BrokenPlugin4', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/INI.pm line 53
    # 	Config::MVP::Reader::INI::INIReader::change_section('Config::MVP::Reader::INI::INIReader=HASH(0x48006390)', 'BrokenPlugin4') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/INI/Reader.pm line 111
    # 	Config::INI::Reader::read_handle('Config::MVP::Reader::INI::INIReader=HASH(0x48006390)', 'IO::File=GLOB(0x480062e8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Mixin/Linewise/Readers.pm line 103
    # 	Mixin::Linewise::Readers::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/INI.pm line 25
    # 	Config::MVP::Reader::INI::read_into_assembler('Config::MVP::Reader::INI=HASH(0x480060d8)', '/home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/tmp/mkPAbY6Wau/source/dist.ini', 'Dist::Zilla::MVP::Assembler::Zilla=HASH(0x47fc2318)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader.pm line 53
    # 	Config::MVP::Reader::read_config('Config::MVP::Reader::INI=HASH(0x480060d8)', '/home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/tmp/mkPAbY6Wau/source/dist.ini', 'HASH(0x47fc2f90)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Reader/Finder.pm line 115
    # 	Config::MVP::Reader::Finder::read_config('Dist::Zilla::MVP::Reader::Finder', 'Path::Tiny=ARRAY(0x47fc2fd8)', 'HASH(0x47fc2f90)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 287
    # 	Dist::Zilla::Dist::Builder::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 103
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x47fc2e58)', 'Try::Tiny::Catch=REF(0x41cc4798)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 312
    # 	Dist::Zilla::Dist::Builder::_load_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47fc2498)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Dist/Builder.pm line 45
    # 	Dist::Zilla::Dist::Builder::from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)') called at /home/cpansand/.cpan/build/2016092509/Dist-Zilla-6.007-N_k_2G/blib/lib/Dist/Zilla/Tester.pm line 196
    # 	Class::MOP::Class:::around('CODE(0x4510e810)', 'Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Class/MOP/Method/Wrapped.pm line 164
    # 	Dist::Zilla::Tester::_Builder::_wrapped_from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/amd64-freebsd/Class/MOP/Method/Wrapped.pm line 95
    # 	Dist::Zilla::Tester::_Builder::from_config('Dist::Zilla::Tester::_Builder', 'HASH(0x47f9cf00)', 'HASH(0x47fc14f8)') called at t/diagnostics/plugin-fail.t line 13
    # 	main::mkconfig('corpus/dist/DZT', 'ARRAY(0x41cc46c0)') called at t/diagnostics/plugin-fail.t line 43
    # 	main::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Fatal.pm line 129
    # 	Test::Fatal::try {...}  at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 101
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Try/Tiny.pm line 94
    # 	Try::Tiny::try('CODE(0x41cc4270)', 'Try::Tiny::Catch=REF(0x47fc1be8)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Fatal.pm line 136
    # 	Test::Fatal::exception('CODE(0x41cfe4f8)') called at t/diagnostics/plugin-fail.t line 43
    # 	main::__ANON__ at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Builder.pm line 300
    # 	eval {...} at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/Builder.pm line 300
    # 	Test::Builder::subtest('Test::Builder=HASH(0x41d08540)', 'BrokenPlugin4', 'CODE(0x41cfe480)') called at /usr/perl5.24.0p/lib/site_perl/5.24.0/Test/More.pm line 807
    # 	Test::More::subtest('BrokenPlugin4', 'CODE(0x41cfe480)') called at t/diagnostics/plugin-fail.t line 48
    # Compilation failed in require at /usr/perl5.24.0p/lib/site_perl/5.24.0/Module/Runtime.pm line 317, <GEN3> line 8. at /usr/perl5.24.0p/lib/site_perl/5.24.0/Config/MVP/Assembler.pm line 144.
    # '
    #     doesn't match '(?^:Some/Package/That/Does/Not/Exist/)'
    # Looks like you failed 2 tests of 3.

#   Failed test 'BrokenPlugin4'
#   at t/diagnostics/plugin-fail.t line 48.
# Looks like you failed 1 test of 7.
t/diagnostics/plugin-fail.t ..... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/7 subtests 
```

Statistical analysis suggests that this happens with Moose 2.1806:
```
****************************************************************
Regression 'mod:Moose'
****************************************************************
Name                   Theta          StdErr     T-stat
[0='const']           1.0000          0.0000    80562839281945184.00
[1='eq_2.1204']       0.0000          0.0000       0.79
[2='eq_2.1209']       0.0000          0.0000       0.91
[3='eq_2.1211']       0.0000          0.0000       0.79
[4='eq_2.1402']       0.0000          0.0000       1.00
[5='eq_2.1403']       0.0000          0.0000       0.79
[6='eq_2.1404']       0.0000          0.0000       0.79
[7='eq_2.1405']       0.0000          0.0000       1.19
[8='eq_2.1603']       0.0000          0.0000       0.91
[9='eq_2.1801']       0.0000          0.0000       1.83
[10='eq_2.1802']              0.0000          0.0000       0.91
[11='eq_2.1803']              0.0000          0.0000       0.97
[12='eq_2.1804']              0.0000          0.0000       2.20
[13='eq_2.1805']              0.0000          0.0000       0.00
[14='eq_2.1806']             -1.0000          0.0000    -72057594037927920.00

R^2= 1.000, N= 75, K= 15
****************************************************************
```

**Comment by karenetheridge** [comment]
I'll investigate and prepare a fix (whether it be in Moose or Dist::Zilla).

**Comment by karenetheridge** [comment]
I think the preferred fix belongs in Moose.. pursuing a fix there.

**Comment by rjbs** [comment]
This has been fixed as of DZ v6.008
